### PR TITLE
hostap: Support get STA connection state

### PR DESCRIPTION
**Pull request by xinyu0322**
hostap: Support getting Wi-Fi connection state

### CODE REVIEW

**Comment by jukkar** [comment]
Isn't this the same as #75250, why do send it again?

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -70,6 +70,8 @@ enum net_request_wifi_cmd {
 	NET_REQUEST_WIFI_CMD_AP_DISABLE,
 	/** Get interface status */
 	NET_REQUEST_WIFI_CMD_IFACE_STATUS,
+	/** Get CONNECTION status */
+	NET_REQUEST_WIFI_CMD_CONNECTION_STATUS,
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:74]: can't we just use `NET_REQUEST_WIFI_CMD_IFACE_STATUS` which is a superset? Having a command for each info is an overkill.

Comment (xinyu0322) [reply include/zephyr/net/wifi_mgmt.h:74]: 1.For NET_REQUEST_WIFI_CMD_IFACE_STATUS, obtaining the connection state is the most commonly used, and other information such as rssi is not commonly used, and there is no need to add a special interface for other information.
2.Sometimes we just need to get the connection state. If we use NET_REQUEST_WIFI_CMD_IFACE_STATUS, we need to call z_wpa_ctrl_status, z_wpa_ctrl_signal_poll, wpa_drv_get_conn_info, etc., as well as the process of supplicant and driver interaction to get a lot of unnecessary information, which is very redundant and complicated. Adding an interface to get the connection state is a simple and clear way.

Comment (krish2718) [reply include/zephyr/net/wifi_mgmt.h:74]: > supplicant and driver interaction to get a lot of unnecessary information, which is very redundant and complicated. Adding an interface to get the connection state is a simple and clear way.

I understand, but if we don't group things then the entire interface will be bloated with commands to get individual pieces, esp. as this interface needs to be implemented by all L2 drivers/NMs.

**Pull request closed without merging**